### PR TITLE
chore: split download.ts to include tests

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
refactoring to include tests before adding podman v5

bump nock to be able to mock octokit that is using fetch API

### What issues does this PR fix or reference?

related to https://github.com/containers/podman-desktop/issues/6375

### How to test this PR?

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: extensions/podman/scripts/podman-download.ts [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+    const disk = qemuData.formats['qcow2.xz'].disk;
+
+    // get the disk location
+    const diskLocation = disk.location;
+
+    // get the sha2556
+    const sha256 = disk.sha256;
+
+    // download the file and check the sha256
+    const destDir = path.resolve(__dirname, '..', 'assets');
```
Comment (lstocchi) [inline extensions/podman/scripts/podman-download.ts]: You can use `this.#assetsFolder`.

File: extensions/podman/scripts/podman-download.ts [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+
+  // For Windows binaries, grab the latest release from GitHub repository
+  async download(repo: string, arch: string): Promise<void> {
+    if (this.#downloadAttempt >= this.MAX_DOWNLOAD_ATTEMPT) {
+      console.error('Max download attempt reached, exiting...');
+      process.exit(1);
+    }
+
+    // download the file and check the sha256
+    const destDir = path.resolve(__dirname, '..', 'assets');
```
Comment (lstocchi) [inline extensions/podman/scripts/podman-download.ts]: You can use this.#assetsFolder

File: extensions/podman/scripts/podman-download.ts [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+      }
+    }
+    if (!msiSha) {
+      console.error(`Can't find SHA256 sum for ${artifactName} in:\n${shaFileContent}`);
+      process.exit(1);
+    }
+
+    if (!fs.existsSync(this.#assetsFolder)) {
+      fs.mkdirSync(this.#assetsFolder);
+    }
```
Comment (lstocchi) [inline extensions/podman/scripts/podman-download.ts]: As we check the existance and create it if there is none in multiple places, couldn't do it in the constructor?

File: extensions/podman/scripts/podman-download.ts [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+      version: string;
+      platform: {
+        win32: { version: string; fileName: string };
+        darwin: {
+          version: string;
+          arch: { x64: { fileName: string }; arm64: { fileName: string }; universal?: { fileName: string } };
+        };
+      };
+    },
+    private airgapSupport: boolean,
```
Comment (lstocchi) [inline extensions/podman/scripts/podman-download.ts:62]: I wonder when this will be false.

Comment (benoitf) [reply extensions/podman/scripts/podman-download.ts:62]: for podman 5, we'll call (no airgap mode for now)

```
const podman5Download = new PodmanDownload(podman5JSON, false);
podman5Download.downloadBinaries();
```

File: extensions/podman/scripts/podman-download.ts (hunk 3 above)
Comment (benoitf) [reply extensions/podman/scripts/podman-download.ts]: I'll check 👍

Comment (benoitf) [reply extensions/podman/scripts/podman-download.ts]: fixed

File: extensions/podman/scripts/podman-download.ts (hunk 2 above)
Comment (benoitf) [reply extensions/podman/scripts/podman-download.ts]: fixed

File: extensions/podman/scripts/podman-download.ts (hunk 1 above)
Comment (benoitf) [reply extensions/podman/scripts/podman-download.ts]: fixed

File: extensions/podman/scripts/podman-download.ts [hunk 5]
```diff
@@ -0,0 +1,427 @@
...
+      request.end();
+    });
+  }
+}
+
+export class PodmanDownloadFcosImage {
+  #downloadAttempt = 0;
+  #httpsDownloader: HttpsDownloader;
+  #shaCheck: ShaCheck;
+  #asssetsFolder: string;
```
Comment (lstocchi) [inline extensions/podman/scripts/podman-download.ts]: typo? on extra `s`

**Review by lstocchi — APPROVED** [review]
LGTM!!

File: extensions/podman/scripts/podman-download.ts (hunk 5 above)
Comment (benoitf) [reply extensions/podman/scripts/podman-download.ts]: 🤦 

fixed

**Pull request merged**